src/shared/azureagent/identities.go

```diff
@@ -99,8 +99,26 @@ func (a *Agent) DeleteUserAssignedIdentity(ctx context.Context, namespace string
 	userAssignedIdentityName := a.GenerateUserAssignedIdentityName(namespace, accountName)
 	federatedIdentityCredentialsName := a.generateFederatedIdentityCredentialsName(namespace, accountName)
 
+	// Delete roles assigned to the identity
+	identity, err := a.FindUserAssignedIdentity(ctx, namespace, accountName)
+	if err != nil {
```

Comment:
you should catch and ignore NotFound errors here, to avoid failing deletion just because the resource is already deleted. See reference a few lines down (`!azureerrors.IsNotFoundErr...`).